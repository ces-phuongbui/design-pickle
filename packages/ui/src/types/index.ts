interface Image {
  id: string;
  documentId: string;
  url: string;
  alternativeText: string | null;
  name: string;
}

export enum ButtonType {
  DEFAULT = "default",
  DESTRUCTIVE = "destructive",
  OUTLINE = "outline",
  SECONDARY = "secondary",
  GHOST = "ghost",
  LINK = "link",
}

export interface IButton {
  label: string;
  href?: string;
  type: ButtonType;
  color?: string;
}

export type ICarouselItem = {
  id: string | number;
  image: Image;
  name: string;
};

export type CardFeature = {
  id: number | string;
  title: string;
  cta: IButton;
  backgroundImage?: Image;
  description?: string;
};

export type CardStat = {
  value: string | number;
  label?: string;
  subLabel: string;
};

export type ComponentType =
  | "section.introduce-header"
  | "section.social-carousel"
  | "section.features"
  | "section.start";

interface Base<T extends ComponentType, D extends {} = {}> {
  __component: T;
  id: string;
  createdAt: string;
  updatedAt: string;
  data?: D;
}

export interface NavLink {
  href: string;
  text: string;
  isExternal: boolean;
  isPrimary: boolean;
}

export type Block = Introduce | SocialCarousel | Features | Stat;

export interface Introduce extends Base<"section.introduce-header"> {
  title: string;
  description: string;
  button: IButton;
}

export interface SocialCarousel extends Base<"section.social-carousel"> {
  bannerImage: Image;
  logos: ICarouselItem[];
}

export interface Features extends Base<"section.features"> {
  title: string;
  subTitle?: string;
  cards: CardFeature[];
}

export interface Stat extends Base<"section.start"> {
  title: string;
  subTitle: string;
  cards: CardStat[];
}
