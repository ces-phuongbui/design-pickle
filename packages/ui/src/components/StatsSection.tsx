import React from "react";
import { StatCard } from "./ui/stat-card";
import { Stat } from "../types";

const StatsSection = ({ title, subTitle, cards }: Stat) => {
  console.log("stat: ", title, subTitle, cards);
  return (
    <section className="bg-black py-16 md:py-24">
      <div className="container mx-auto px-4">
        <div className="mx-auto mb-12 max-w-3xl text-center">
          <h2 className="mb-4 text-3xl font-bold text-white md:text-4xl">
            {title}
          </h2>
          <p className="text-gray-400">{subTitle}</p>
        </div>

        <div className="mx-auto grid max-w-5xl gap-4 sm:grid-cols-2 lg:grid-cols-4">
          {cards.map((card, index) => (
            <StatCard
              key={index}
              value={card.value}
              label={card.label}
              subLabel={card.subLabel ?? ""}
            />
          ))}
        </div>
      </div>
    </section>
  );
};

export default StatsSection;
