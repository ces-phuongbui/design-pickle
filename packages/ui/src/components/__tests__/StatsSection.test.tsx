import { Stat } from "@repo/ui/types/index";
import { render, screen } from "@testing-library/react";
import StatsSection from "../StatsSection";

describe("StatsSection", () => {
  const mockProps: Stat = {
    __component: "section.start",
    id: "1",
    createdAt: "2023-10-01T00:00:00.000Z",
    updatedAt: "2023-10-01T00:00:00.000Z",
    title: "Our Stats",
    subTitle: "Check out our amazing statistics",
    cards: [
      {
        value: "100+",
        label: "Projects",
        subLabel: "Completed",
      },
      {
        value: "50+",
        label: "Clients",
        subLabel: "Satisfied",
      },
      {
        value: "10+",
        label: "Years",
        subLabel: "Experience",
      },
      {
        value: "24/7",
        label: "Support",
        subLabel: "Available",
      },
    ],
  };

  it("renders correctly and matches snapshot", () => {
    const { container } = render(<StatsSection {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the title and subtitle correctly", () => {
    render(<StatsSection {...mockProps} />);
    const title = screen.getByText("Our Stats");
    const subTitle = screen.getByText("Check out our amazing statistics");
    expect(title).toBeInTheDocument();
    expect(subTitle).toBeInTheDocument();
  });

  it("renders all stat cards correctly", () => {
    render(<StatsSection {...mockProps} />);
    const statCards = screen.getAllByTestId("stat-card");
    expect(statCards.length).toBe(4);

    // Check the content of the first card
    expect(screen.getByText("100+")).toBeInTheDocument();
    expect(screen.getByText("Projects")).toBeInTheDocument();
    expect(screen.getByText("Completed")).toBeInTheDocument();

    // Check the content of the second card
    expect(screen.getByText("50+")).toBeInTheDocument();
    expect(screen.getByText("Clients")).toBeInTheDocument();
    expect(screen.getByText("Satisfied")).toBeInTheDocument();
  });

  it("renders stat cards without optional subLabel", () => {
    const propsWithoutSubLabel: Stat = {
      ...mockProps,
      cards: [
        {
          value: "100+",
          label: "Projects",
          subLabel: "",
        },
      ],
    };
    render(<StatsSection {...propsWithoutSubLabel} />);
    const statCard = screen.getByTestId("stat-card");
    expect(statCard).toBeInTheDocument();
    expect(screen.queryByText("Completed")).toBeNull();
  });
});
