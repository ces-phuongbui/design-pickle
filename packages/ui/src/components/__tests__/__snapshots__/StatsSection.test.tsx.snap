// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsSection renders correctly and matches snapshot 1`] = `
<div>
  <section
    class="bg-black py-16 md:py-24"
  >
    <div
      class="container mx-auto px-4"
    >
      <div
        class="mx-auto mb-12 max-w-3xl text-center"
      >
        <h2
          class="mb-4 text-3xl font-bold text-white md:text-4xl"
        >
          Our Stats
        </h2>
        <p
          class="text-gray-400"
        >
          Check out our amazing statistics
        </p>
      </div>
      <div
        class="mx-auto grid max-w-5xl gap-4 sm:grid-cols-2 lg:grid-cols-4"
      >
        <div
          class="flex flex-col items-center space-y-2 rounded-lg bg-zinc-900 p-6 text-center text-white"
          data-testid="stat-card"
        >
          <div
            class="text-3xl font-bold md:text-4xl lg:text-5xl"
          >
            100+
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="text-base text-gray-200 md:text-lg font-bold"
            >
              Projects
            </div>
            <div
              class="text-sm text-gray-400"
            >
              Completed
            </div>
          </div>
        </div>
        <div
          class="flex flex-col items-center space-y-2 rounded-lg bg-zinc-900 p-6 text-center text-white"
          data-testid="stat-card"
        >
          <div
            class="text-3xl font-bold md:text-4xl lg:text-5xl"
          >
            50+
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="text-base text-gray-200 md:text-lg font-bold"
            >
              Clients
            </div>
            <div
              class="text-sm text-gray-400"
            >
              Satisfied
            </div>
          </div>
        </div>
        <div
          class="flex flex-col items-center space-y-2 rounded-lg bg-zinc-900 p-6 text-center text-white"
          data-testid="stat-card"
        >
          <div
            class="text-3xl font-bold md:text-4xl lg:text-5xl"
          >
            10+
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="text-base text-gray-200 md:text-lg font-bold"
            >
              Years
            </div>
            <div
              class="text-sm text-gray-400"
            >
              Experience
            </div>
          </div>
        </div>
        <div
          class="flex flex-col items-center space-y-2 rounded-lg bg-zinc-900 p-6 text-center text-white"
          data-testid="stat-card"
        >
          <div
            class="text-3xl font-bold md:text-4xl lg:text-5xl"
          >
            24/7
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="text-base text-gray-200 md:text-lg font-bold"
            >
              Support
            </div>
            <div
              class="text-sm text-gray-400"
            >
              Available
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
